src/udev/udevd.c

```diff
@@ -1577,7 +1577,8 @@ static int manager_new(Manager **ret, int fd_ctrl, int fd_uevent, const char *cg
         if (r < 0)
                 return log_error_errno(r, "Failed to initialize device monitor: %m");
 
-        (void) sd_device_monitor_set_receive_buffer_size(manager->monitor, 128 * 1024 * 1024);
+        if (fd_uevent < 0)
```

Comment:
looks good. Could you add a comment here for the rationale though? This is not obvious to grok otherwise

```
/* Bump receiver buffer, but only if we are not called via socket activation, as in that case systemd sets the receive buffer size for us, and the value in the .socket unit should take full effect. */
```

or something like that.